### PR TITLE
Refactor Ratings package to allow unit testing of functions

### LINKED ISSUE #3136: Refactor ratings package to allow unit-testing
## **Description**:

Test coverage of the Platform needs to be increased. I will refactor the *ratings* package to allow it to be tested properly.

At this point, rewriting the codebase is not practical. My solution will only need a small refactoring of how the functions are declared so they can have isolated scopes and can be tested outside the Meteor environment.

## **Please note**

This is an experiment. If the refactor is successful and the tests can be performed without compromising the current codebase, we can increase the test coverage of the platform to 100% in a matter of months.

### PR DESCRIPTION
**Pull request by midyan**
Closes #3136

## Changes
Describe your changes here as a bulleted list:
- change function declaration stategy
- change folder structure

## Developer checklist
This checklist is to be completed by the PR developer:

- [ ] Alternative solutions were compared/discussed before writing code
    - [ ] trade-offs with this solution are considered acceptable
- [ ] Code in this PR adheres to the [project styleguide](CONTRIBUTING.md)
- [ ] This pull request does not decrease project test coverage
- [ ] If the code changes existing database collection(s), migration has been written
- [ ] If UI texts are added or changed, all texts are internationalized

## Reviewer checklist
Reviewed by: @mauriciovieira

This list is to be completed by the pull request reviewer:
- [ ] Code works as described/expected
- [ ] Code seems to be error free
    - [ ] no browser console errors visible
    - [ ] no server console errors visible
    - [ ] passes CI build
- [ ] Code is written in a way that promotes maintainability
    - [ ] easy to understand
    - [ ] well organized
    - [ ] follows project coding standards and conventions
    - [ ] well documented

### CODE REVIEW

File: apinf_packages/ratings/collection/permissions.js [hunk 1]
```diff
@@ -6,12 +6,10 @@ https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence
...
-    return !!(userId);
-  },
-  update (userId, rating) {
-    return (userId === rating.userId);
-  },
-});
+// Import functions
+import collectionFunction from './functions.js';
+
+// Deconstruct relevante functions
```
Comment (mauriciovieira) [inline apinf_packages/ratings/collection/permissions.js]: s/relevante/relevant/

File: apinf_packages/ratings/collection/server/publications.js [hunk 2]
```diff
@@ -3,50 +3,54 @@ This file is covered by the EUPL license.
 You may obtain a copy of the licence at
 https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence-eupl-v11 */
 
+/* eslint-disable meteor/audit-argument-checks */
```
Comment (mauriciovieira) [inline apinf_packages/ratings/collection/server/publications.js]: Any `eslint-disable` for the whole file is not a good idea...

Why do you need this? And why not use the inline notation (eslint-disable-next-line) ?

File: tests/apinf_packages/ratings/ratings.test.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
+/* Copyright 2017 Apinf Oy
+This file is covered by the EUPL license.
+You may obtain a copy of the licence at
+https://joinup.ec.europa.eu/community/eupl/og_page/european-union-public-licence-eupl-v11 */
+
+/* eslint-env mocha */
+/* eslint-disable no-unused-expressions */
```
Comment (mauriciovieira) [inline tests/apinf_packages/ratings/ratings.test.js]: Probably there are workarounds to not disable eslint on this file.

File: tests/apinf_packages/ratings/ratings.test.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+          // Flag to define if user and update rating
+          const canUpdate = update(userId, rating);
+
+          expect(canUpdate).toEqual(false);
+        });
+      });
+    });
+
+    describe('publications.js functions', () => {
+      describe('publishMyApiBackendRating fucntion', () => {
```
Comment (mauriciovieira) [inline tests/apinf_packages/ratings/ratings.test.js]: typo on fucntion.

File: tests/apinf_packages/ratings/ratings.test.js [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+          const result =
+            publishMyApiBackendRating({ check, ApiBackendRatings, context })(apiBackendId);
+
+          expect(result.userId).toEqual(userId);
+          expect(result.apiBackendId).toEqual(apiBackendId);
+          expect(Number.isInteger(result.rating)).toEqual(true);
+        });
+      });
+
+      describe('publishMyApiBackendRatings fucntion', () => {
```
Comment (mauriciovieira) [inline tests/apinf_packages/ratings/ratings.test.js]: Typo on fucntion

File: tests/apinf_packages/ratings/ratings.test.js [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+
+          const result =
+            publishMyApiBackendRatings({ ApiBackendRatings, context })();
+
+          expect(Array.isArray(result)).toEqual(true);
+          expect(result.length).toEqual(2);
+        });
+      });
+
+      describe('publishApiBackendRatings fucntion', () => {
```
Comment (mauriciovieira) [inline tests/apinf_packages/ratings/ratings.test.js]: typo on fuction

**Comment by marla-singer** [comment]
@preriasusi What to do with this PR?

**Comment by 55** [comment]
@preriasusi I would suggest to stop/close work on this, since rating component in it's current form will be deprecated later this year as a part of ongoing redesign process.

**Pull request closed without merging**